symphony-bdk-core/src/test/java/com/symphony/bdk/core/service/MessageServiceTest.java

```diff
@@ -0,0 +1,237 @@
+
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.symphony.bdk.core.api.invoker.ApiException;
+import com.symphony.bdk.core.api.invoker.ApiRuntimeException;
+import com.symphony.bdk.core.auth.AuthSession;
+import com.symphony.bdk.gen.api.AttachmentsApi;
+import com.symphony.bdk.gen.api.DefaultApi;
+import com.symphony.bdk.gen.api.MessageApi;
+import com.symphony.bdk.gen.api.MessageSuppressionApi;
+import com.symphony.bdk.gen.api.MessagesApi;
+import com.symphony.bdk.gen.api.PodApi;
+import com.symphony.bdk.gen.api.StreamsApi;
+import com.symphony.bdk.gen.api.model.MessageIdsFromStream;
+import com.symphony.bdk.gen.api.model.MessageMetadataResponse;
+import com.symphony.bdk.gen.api.model.MessageReceiptDetailResponse;
+import com.symphony.bdk.gen.api.model.MessageStatus;
+import com.symphony.bdk.gen.api.model.MessageSuppressionResponse;
+import com.symphony.bdk.gen.api.model.V4Message;
+import com.symphony.bdk.gen.api.model.V4Stream;
+import com.symphony.bdk.template.api.TemplateEngine;
+import com.symphony.bdk.template.api.TemplateException;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.mockito.junit.jupiter.MockitoSettings;
+import org.mockito.quality.Strictness;
+
+import java.util.Collections;
+
+/**
+ * Test class for the {@link MessageService}.
+ */
+@ExtendWith(MockitoExtension.class)
+@MockitoSettings(strictness = Strictness.LENIENT)
+class MessageServiceTest {
+
+  private static final String MESSAGE = "message";
+  private static final String STREAM_ID = "streamId";
+  private static final String MESSAGE_ID = "messageId";
+  private static final String TEMPLATE_NAME = "template";
+
+  @Mock
+  private MessagesApi messagesApi;
+
+  @Mock
+  private MessageApi messageApi;
+
+  @Mock
+  private MessageSuppressionApi messageSuppressionApi;
+
+  @Mock
+  private StreamsApi streamsApi;
+
+  @Mock
+  private PodApi podApi;
+
+  @Mock
+  private AttachmentsApi attachmentsApi;
+
+  @Mock
+  private DefaultApi defaultApi;
+
+  @Mock
+  private AuthSession authSession;
+
+  @Mock
+  private TemplateEngine templateEngine;
+
+  private MessageService messageService;
+
+  @BeforeEach
+  void setUp() {
+    messageService = new MessageService(messagesApi, messageApi, messageSuppressionApi, streamsApi, podApi,
+        attachmentsApi, defaultApi, authSession, templateEngine);
+    when(authSession.getSessionToken()).thenReturn("sessionToken");
+    when(authSession.getKeyManagerToken()).thenReturn("keyManagerToken");
+  }
+
+  @Test
+  void testGetMessages() throws ApiException {
```

Comment:
we don't test much here, I would perhaps pass different values for args and check the values with eq() just to avoid interverting parameters for instance

Reply:
> we don't test much here, I would perhaps pass different values for args and check the values with eq() just to avoid interverting parameters for instance

Yes indeed. Fact is, we don't have much logic in the MessageService class apart from forwarding parameters and parse the returned json from the HTTP calls.